up_client_socket/rust/u_transport_socket.rs

```diff
@@ -0,0 +1,395 @@
+/*
+ * Copyright (c) 2024 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use async_std::io;
+use async_trait::async_trait;
+use tokio::runtime::Runtime;
+use up_rust::UListener;
+use up_rust::{UAttributesValidators, UriValidator};
+use up_rust::{UCode, UMessage, UMessageType, UStatus, UTransport, UUri};
+
+use protobuf::Message;
+use std::net::TcpStream as TcpStreamSync;
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
```

Comment:
Advise that when working with Rust async code to use the `Mutex` implementation from your async framework.

In this case if you're using tokio it's under `use tokio::sync::Mutex;` or if async-std it's under `use async_std::sync::Mutex`.

This is important so that you don't get in trouble with the borrow checker for holding a lock past an `.await` poitn.